File: Business logic tier/API/Data/IRecipeService.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using API.Models;

namespace API.Data
{
    public interface IRecipeService
    {
        Task<Recipe> AddRecipe(Recipe recipe);
        Task<Recipe> GetRecipe(int id);
        Task<IList<Recipe>> GetAllRecipes();
        Task<Recipe> UpdateRecipe(Recipe recipe);
        Task<Recipe> RemoveRecipe(Recipe recipe);
        Recipe DataToBusiness(DataRecipe dataRecipe);
        DataRecipe BusinessToData(Recipe recipe);

    }
}

File: Data tier/Java/src/main/java/via/sep3/demo/persistance/IRecipeService.java
package via.sep3.demo.persistance;

import via.sep3.demo.Model.Recipe;

import java.util.List;

public interface IRecipeService {

    List<Recipe> GetAllRecipes();
    Recipe GetRecipe(String Name);
    Recipe AddRecipe(Recipe recipe);
    Recipe RemoveRecipe(Recipe recipe);
    Recipe UpdateRecipe(Recipe recipe);
}
